build_tools/fetch_artifacts.py

```diff
@@ -31,24 +35,52 @@ def s3_bucket_exists(run_id):
     return process.returncode == 0
 
 
-def s3_exec(variant, package, run_id, build_dir):
-    cmd = [
-        "aws",
-        "s3",
-        "cp",
-        f"s3://therock-artifacts/{run_id}/{package}_{variant}.tar.xz",
-        build_dir,
-        "--no-sign-request",
-    ]
-    log(f"++ Exec [{cmd}]")
+def s3_commands_for_artifacts(artifacts, run_id, build_dir, variant):
+    """
+    This function collects AWS S3 copy commands to execute later in parallel, where
+    the commands copy TheRock artifacts from the S3 bucket to an output directory
+    """
+    cmds = []
+    for artifact in artifacts:
+        cmds.append(
+            [
+                "aws",
+                "s3",
+                "cp",
+                f"s3://therock-artifacts/{run_id}/{artifact}_{variant}.tar.xz",
+                build_dir,
+                "--no-sign-request",
+                "--quiet",
+            ]
+        )
+    return cmds
+
+
+def subprocess_run(cmd):
+    """
+    This function runs a command via subprocess run
+    """
```

Comment:
This docstring is pretty trivial and redundant with the call signature. This could be more useful by adding a type to `cmd` (string? list of strings?) and documenting the return value and possible exceptions.

In particular... I find it odd to _return_ a string like `++ Exec completed: `. How about instead logging what command is going to be run _first_, running the command, logging when the command completes, and then returning without a result?

You could also just take this function as is: https://github.com/ROCm/TheRock/blob/fc0fabcecab0227a1623141598c169d2f17df16c/build_tools/fetch_sources.py#L27-L31

If this needs to work a certain way (returning a string with the result) for `executor.submit(`, _that_ should be documented in the function name and/or the docstring.